### PR TITLE
fix #224 Unable to set multiple cookies in a single web request

### LINKED ISSUE #224: Unable to set multiple cookies in a single web request
According to [RFC6265 section 3](https://tools.ietf.org/html/rfc6265#section-3), the server should be able to set multiple cookies in a single web request.

> An origin server can include multiple Set-Cookie header fields in a single response.
...
> Origin servers SHOULD NOT fold multiple Set-Cookie header fields into a single header field.

`jester.nim#setCookie()` uses `setHeader()` which overwrites any previous Set-Cookie header.

### PR DESCRIPTION
**Pull request by itsumura-h**
fix #224 Unable to set multiple cookies in a single web request
multiple cookies should be allowed

### CODE REVIEW

File: jester.nim [hunk 1]
```diff
@@ -516,11 +516,12 @@ template setHeader(headers: var Option[RawHeaders], key, value: string): typed =
     block outer:
       # Overwrite key if it exists.
       var h = headers.get()
-      for i in 0 ..< h.len:
-        if h[i][0] == key:
-          h[i][1] = value
-          headers = some(h)
-          break outer
+      if key != "Set-cookie": # multiple cookies should be allowed
```
Comment (dom96) [inline jester.nim]: I wonder if it makes sense to change semantics just for Set-Cookie, would it make sense to allow multiple header values for all keys?

Comment (itsumura-h) [reply jester.nim]: Seeing existing code, I thought your design aims for not allowing multiple header for same key. However at least `Set-cookie` should allow multiple header.
Would it be better to allow multiple header for any key?

Comment (Araq) [reply jester.nim]: Just allow multiple headers for every key and accept the fact that HTTP is just a bunch of silly bytes with no underlying design.

Comment (dom96) [reply jester.nim]: @itsumura-h I think so, I'd rather be consistent here.

**Comment by itsumura-h** [comment]
@dom96 
I updated to allow multiple header whatever key is

**Comment by itsumura-h** [comment]
It works like this

```nim
import jester, httpcore

routes:
  get "/":
    let headers = @[
      ("aaa", "111"),
      ("aaa", "222"),
      ("aaa", "333"),
      ("Set-cookie", "aaa"),
      ("Set-cookie", "bbb"),
    ]
    resp Http200, headers, "test"
```

response header
```
aaa: 111, 222, 333
Content-Length: 4
Date: Mon, 13 Apr 2020 22:10:36 GMT
Server: HttpBeast
Set-cookie: aaa
Set-cookie: bbb
```

**Comment by dom96** [comment]
I still don't want the behaviour of `setHeader` to change. If you want `resp` to work like this then make it call a new addHeader template.

**Pull request closed without merging**